frontend/types/api/notifications.ts

```diff
@@ -0,0 +1,176 @@
+/**
+ * ------------------------------------------------------------
+ * Clients Table
+ */
+export interface NotificationsClientsTableRow {
+  client_name: string;
+  version: string;
+  timestamp: number /* uint64 */;
+}
+export type InternalGetNotificationClients = ApiPagingResponse<NotificationsClientsTableRow>;
+/**
+ * ------------------------------------------------------------
+ * Rocket Pool Table
+ */
+export interface NotificationRocketPoolTableRow {
+  timestamp: number /* uint64 */;
+  event_type: string;
+  alert_value?: number /* float64 */; // only for some notification types, e.g. max collateral
+  node_address: Hash;
+}
+export type InternalGetNotificationRocketPool = ApiPagingResponse<NotificationRocketPoolTableRow>;
+/**
+ * ------------------------------------------------------------
+ * Networks Table
+ */
+export interface NotificationNetworksTableRow {
+  chain_id: number /* uint64 */;
+  timestamp: number /* uint64 */;
+  event_type: string;
+  alert_value: string /* decimal.Decimal */; // wei string for gas alerts, otherwise percentage (0-1) for participation rate
+}
+export type InternalGetNotificationNetworks = ApiPagingResponse<NotificationNetworksTableRow>;
+/**
+ * ------------------------------------------------------------
+ * Notification Settings
+ */
+export interface NotificationEventsNetwork {
+  gas_above: string /* decimal.Decimal */;
+  gas_below: string /* decimal.Decimal */;
+  participation_rate: number /* float64 */;
+}
+export interface NotificationSettingsNetwork {
+  chain_id: number /* uint64 */;
+  subscribed_events: NotificationEventsNetwork;
+}
+export type InternalPutNotificationSettingsNetworksResponse = ApiDataResponse<NotificationSettingsNetwork>;
+export interface NotificationSettingsPairedDevice {
+  id: string;
+  name?: string;
+  enable_notifications: boolean;
+}
+export type InternalPutNotificationSettingsPairedDevicesResponse = ApiDataResponse<NotificationSettingsNetwork>;
+export interface NotificationEventsGeneral {
+  machines_offline: boolean;
+  machines_storage_usage: number /* float64 */;
+  machines_cpu_usage: number /* float64 */;
+  machines_memory_usage: number /* float64 */;
+  clients: string[];
+  clients_rocket_pool_smart_node: boolean;
+  clients_mev_boost: boolean;
+  rocket_pool_new_reward_round: boolean;
+  rocket_pool_max_collateral: number /* float64 */;
+  rocket_pool_min_collateral: number /* float64 */;
+}
+export interface NotificationSettings {
+  do_not_disturb_timestamp: number /* uint64 */;
+  enable_email: boolean;
+  enable_push: boolean;
+  subscribed_events: NotificationEventsGeneral;
+  networks: NotificationSettingsNetwork[];
+  paired_devices: NotificationSettingsPairedDevice[];
+}
+export type InternalGetNotificationSettingsResponse = ApiDataResponse<NotificationSettings>;
+export interface NotificationEventsValidatorDashboard {
+  validator_offline: boolean;
+  group_offline: number /* float64 */;
```

Comment:
As JSON does not support `NaN`, can you make tygo generate `number|null` here?
`null` is when the user hasn't set a value (the user never opened the setting dialog or has left the input field empty)